### PR TITLE
Reorganize projects

### LINKED ISSUE #26: Merge delegate and accessible into core
I would propose merging those two as they both have no dependencies. This would also improve the user experience as only one coordinate needs to be specified.
I would keep mockable in its own module, but maybe name it test-macros or similiar

@ioleo what do you think?

### PR DESCRIPTION
**Pull request by ioleo**
Fixes #26 

Also includes minor tweaks:
- updated readme
- added warning suppression via silencer plugin

### CODE REVIEW

**Comment by ioleo** [comment]
Rebased to include `delegate/patch`

File: build.sbt [hunk 1]
```diff
@@ -49,22 +49,14 @@ lazy val root = project
...
-    mock.js,
-    mockExamples.jvm,
-    mockExamples.js,
-    delegate.jvm,
-    delegate.js,
-    delegateTests.jvm,
-    delegateTests.js,
-    delegateExamples.jvm,
-    delegateExamples.js
+    coreExamples.jvm,
```
Comment (mschuwalow) [inline build.sbt:54]: I believe coreTests are missing here

File: core/shared/src/main/scala/zio/macros/core/ModulePattern.scala [hunk 2]
```diff
@@ -79,6 +80,7 @@ private[macros] trait ModulePattern {
       case _ => abort("Module trait and companion object pair not found")
     }
 
+  @silent("pattern var [^\\s]+ in method unapply is never used")
```
Comment (mschuwalow) [inline core/shared/src/main/scala/zio/macros/core/ModulePattern.scala:83]: nice

**Review by mschuwalow — CHANGES_REQUESTED** [review]
Looks good. Only one small remark.

Thanks for taking care of this!

**Pull request merged**